### PR TITLE
discord: add main menu status option

### LINKED ISSUE #12940: Discord: Option to not show game status when on login screen
Keep having misunderstandings where people think I'm playing when I actually just left the client open in the background not logged in, would love if it removed runelite from my status on the login screen.

### PR DESCRIPTION
**Pull request by dillydill123**
Closes #12940

When the client starts up, there is still around ~15 seconds where it will show the status as "In Menu". Afterwards it will go away. I'm not sure how to solve this as it doesn't seem to be related to the startup of the discord plugin. Turning the plugin on and off does not reproduce this behavior. It is only on client startup.

### CODE REVIEW

**Comment by dillydill123** [comment]
I was using the wrong game state. It is correctly not showing up on client startup now.

File: runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java [hunk 1]
```diff
@@ -372,6 +372,12 @@ private void resetState()
 
 	private void checkForGameStateUpdate()
 	{
+		if (!config.showMainMenu() && client.getGameState() != GameState.LOGGED_IN)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java:375]: Just call `client.getGameState()` once and then use it here + also down below.

File: runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordConfig.java [hunk 2]
```diff
@@ -72,11 +72,22 @@ default int actionTimeout()
 		return 5;
 	}
 
+	@ConfigItem(
+			keyName = "showMainMenu",
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordConfig.java:76]: Fix the indenting here.

**Comment by dillydill123** [comment]
I changed the logic because there was a duplicate call to `resetState()`. Everywhere  `checkForGameStateUpdate()` is called, `resetState()` is called right before it.

**Pull request merged**